programs/server/play.html

```diff
@@ -0,0 +1,437 @@
+        }
+
+        th
+        {
+            padding: 0.25rem 0.5rem;
+            text-align: middle;
+            background-color: var(--table-header-color);
+            border: 1px solid var(--border-color);
+        }
+
+        /* The row under mouse pointer is highlight for better legibility. */
+        tr:hover, tr:hover td
+        {
+            background-color: var(--table-hover-color);
+        }
+
+        tr:hover
+        {
+            box-shadow: 0 0 1rem rgba(0, 0, 0, 0.1);
+        }
+
+        #error
+        {
+            background: var(--error-color);
+            white-space: pre-wrap;
+            padding: 0.5rem 1rem;
+            display: none;
+        }
+
+        /* When mouse pointer is over table cell, will display full text (with wrap) instead of cut.
+           TODO Find a way to make it work on touch devices. */
+        td.left:hover
+        {
+            white-space: pre-wrap;
+        }
+
+        /* The style for SQL NULL */
+        .null
+        {
+            color: var(--null-color);
+        }
+    </style>
+</head>
+
+<body>
+    <div id="inputs">
+        <input class="monospace shadow" id="url" type="text" value="http://localhost:8123/" /><input class="monospace shadow" id="user" type="text" value="default" /><input class="monospace shadow" id="password" type="password" />
+    </div>
+    <div>
+        <textarea autofocus spellcheck="false" class="monospace shadow" id="query"></textarea>
+    </div>
+    <div id="run_div">
+        <button class="shadow" id="run">Run</button>
+        <span class="hint">&nbsp;(Ctrl+Enter)</span>
+        <span id="stats"></span>
+        <span id="toggle-dark">🌑</span><span id="toggle-light">🌞</span>
+    </div>
+    <div id="data_div">
+        <table class="monospace shadow" id="data-table"></table>
+        <pre class="monospace shadow" id="data-unparsed"></pre>
+    </div>
+    <p id="error" class="monospace shadow">
+    </p>
+</body>
+
+<script type="text/javascript">
+
+    /// Substitute the address of the server where the page is served.
+    if (location.protocol != 'file:') {
+        document.getElementById('url').value = location.origin;
+    }
+
+    function post()
+    {
+        var url = document.getElementById('url').value +
+            /// Ask server to allow cross-domain requests.
+            '?add_http_cors_header=1' +
+            '&user=' + encodeURIComponent(document.getElementById('user').value) +
+            '&password=' + encodeURIComponent(document.getElementById('password').value) +
+            '&default_format=JSONCompact' +
+            /// Safety settings to prevent results that browser cannot display.
+            '&max_result_rows=1000&max_result_bytes=10000000&result_overflow_mode=break';
+
+        var query = document.getElementById('query').value;
+        var xhr = new XMLHttpRequest;
+
+        xhr.open('POST', url, true);
+        xhr.send(query);
+
+        xhr.onreadystatechange = function()
```

Comment:
I'm curious, any particular reason you're mixing modern CSS features (`var`) with old IE-style JS? (XHR+`onreadystatechange`, `for..in`, `event.charCode` etc)

Reply:
I "learned" HTML, JS and CSS during my middle-school years in early 2000s (actually never read a book).
I continue to mix my old habits with copy-pasting code from Stackoverflow.

Reply:
I tried to use Fetch API but it was not working from `file://` page (most likely due to cross-origin policy, but setting parameter to disable it did not help).

Reply:
I have intention to present this simple page as an example of small self-contained code, so everyone can suggest changes to help me to make it perfect from every kind of sense.